qiskit/visualization/pulse_v2/types.py

```diff
@@ -14,88 +14,189 @@
+PulseInstruction.dt.__doc__ = 'System cycle time.'
+PulseInstruction.frame.__doc__ = 'A reference frame to run instruction.'
+PulseInstruction.inst.__doc__ = 'Pulse instruction.'
 
 
-NonPulseTuple = NamedTuple(
-    'NonPulseTuple',
+BarrierInstruction = NamedTuple(
+    'Barrier',
     [('t0', int),
      ('dt', Optional[float]),
-     ('inst', Union[pulse.Instruction, List[pulse.Instruction]])])
+     ('channels', List[pulse.channels.Channel])]
+)
+BarrierInstruction.__doc__ = 'Data to represent special pulse instruction of barrier.'
+BarrierInstruction.t0.__doc__ = 'A time when the instruction is issued.'
+BarrierInstruction.dt.__doc__ = 'System cycle time.'
+BarrierInstruction.channels.__doc__ = 'A list of channel associated with this barrier.'
 
 
-ChannelTuple = NamedTuple(
-    'ChannelTuple',
-    [('channel', pulse.channels.Channel),
-     ('scaling', float)])
+SnapshotInstruction = NamedTuple(
+    'Snapshots',
+    [('t0', int),
+     ('dt', Optional[float]),
+     ('inst', pulse.instructions.Snapshot)]
+)
+SnapshotInstruction.__doc__ = 'Data to represent special pulse instruction of snapshot.'
+SnapshotInstruction.t0.__doc__ = 'A time when the instruction is issued.'
+SnapshotInstruction.dt.__doc__ = 'System cycle time.'
+SnapshotInstruction.inst.__doc__ = 'Snapshot instruction.'
+
+
+ChartAxis = NamedTuple(
+    'ChartAxis',
+    [('name', str),
+     ('channels', List[pulse.channels.Channel])]
+)
+ChartAxis.__doc__ = 'Data to represent an axis information of chart.'
+ChartAxis.name.__doc__ = 'Name of chart.'
+ChartAxis.channels.__doc__ = 'Channels associated with chart.'
 
 
 ComplexColors = NamedTuple(
     'ComplexColors',
     [('real', str),
      ('imaginary', str)])
+ComplexColors.__doc__ = 'Data to represent a set of color codes for real and imaginary part.'
+ComplexColors.real.__doc__ = 'Color code of real part.'
+ComplexColors.imaginary.__doc__ = 'Color code of imaginary part.'
+
+
+ParsedInstruction = NamedTuple(
+    'ParsedWaveform',
+    [('xvals', np.ndarray),
+     ('yvals', np.ndarray),
+     ('meta', Dict[str, Any])]
+)
+ParsedInstruction.__doc__ = 'Data to represent a parsed pulse instruction for object generation.'
+ParsedInstruction.xvals.__doc__ = 'Numpy array of x axis data.'
+ParsedInstruction.yvals.__doc__ = 'Numpy array of y axis data.'
+ParsedInstruction.meta.__doc__ = 'Dictionary containing instruction details.'
+
+
+HorizontalAxis = NamedTuple(
+    'HorizontalAxis',
+    [('window', Tuple[int, int]),
+     ('axis_map', Dict[int, int]),
+     ('axis_break_pos', List[int]),
+     ('label', str)]
+)
+HorizontalAxis.__doc__ = "Data to represent configuration of horizontal axis."
+HorizontalAxis.window.__doc__ = "Left and right edge of graph."
+HorizontalAxis.axis_map.__doc__ = 'Mapping of apparent coordinate system and actual location.'
+HorizontalAxis.axis_break_pos.__doc__ = "Locations of axis break."
+HorizontalAxis.label.__doc__ = "Label of horizontal axis."
+
+
+class DrawingWaveform(str, Enum):
```

Comment:
Why do these need to inherit `str`?

Reply:
Because the drawing object hash is partly generated from this information I need something hashable. Can I remove this inheritance?

Reply:
What type is `DrawingWaveform.REAL`? Is This a string, or something else?